server/base-routes.js

```diff
@@ -164,15 +164,18 @@ var baseRoutes = [
       var features = request.query.features + ',Intl.~locale.' + locale;
       var flags = request.query.flags ? request.query.flags.split(',') : [];
 
-      var polyfills = PolyfillSet.fromQueryParam(features || 'default', flags);
+      var polyfills = PolyfillSetFromQueryParams(features || 'default', flags);
       var params = {
         features: polyfills.get(),
         minify: true,
         unknown: 'polyfill'
       };
       params.uaString = request.plugins.scooter.source;
-      polyfillio.getPolyfillString(params).then(function(bundleString) {
-        reply(bundleString).type('application/javascript; charset=utf-8').vary('User-Agent');
+
+      return new Promise((resolve) => {
```

Comment:
I'm guessing new hapi just expects things to be handled as promises now?

Reply:
yeah, if it can't return a value synchronously, you return a promise for the value, or use async/await (so... promises, basically)